src/middlewared/middlewared/plugins/kubernetes_linux/update.py

```diff
@@ -86,20 +86,24 @@ async def license_active(self):
         return can_run_apps
 
     @private
-    def check_config_on_apps_dataset(self, pool, verrors, force, schema_name):
-        apps_ds = applications_ds_name(pool)
-        if force or not self.middleware.call_sync(
-            'zfs.dataset.query', [['id', '=', apps_ds]], {'extra': {'retrieve_children': False}}
-        ):
-            return
-
+    def missing_apps_dataset(self, apps_ds):
         missing = []
         for ds_name in self.middleware.call_sync('kubernetes.kubernetes_datasets', apps_ds):
             if not self.middleware.call_sync(
                 'zfs.dataset.query', [['id', '=', ds_name]], {'extra': {'retrieve_children': False}}
```

Comment:
Since we're just checking for dataset existence from the look of it, this can be optimized to not retrieve properties.

Reply:
We might also be able to do this in a single dataset query, right?